src/components/visiumQC/Cleanup.tsx

```diff
@@ -0,0 +1,119 @@
+
+  /**Initialise all comments for slots when labware changes**/
+  React.useEffect(() => {
+    if (!labware) return;
+    setSlotComments([]);
+  }, [labware, setFieldValue]);
+
+  /**Update form values when ever comments associated with slots change**/
+  React.useEffect(() => {
+    setFieldValue('slotComments', slotComments);
+  }, [slotComments, setFieldValue]);
+
+  const getComment = (address: string) => {
+    const slotComment = slotComments.find((sc) => address === sc.address);
+    return slotComment ? slotComment.commentId : '';
+  };
+
+  const handleCommentChange = React.useCallback(
+    (address: string, commentIdVal: string) => {
+      const commentId = commentIdVal !== '' ? Number(commentIdVal) : undefined;
+      setSlotComments((prevSlotComments: AddressCommentInput[]) => {
+        const slotComments = [...prevSlotComments];
+        const findIndex = slotComments.findIndex((sc) => sc.address === address);
+        if (findIndex >= 0) {
+          slotComments.splice(findIndex, 1);
+        }
+        if (!commentId) {
+          return [...slotComments];
+        } else return [...slotComments, { address, commentId }];
+      });
+    },
+    [setSlotComments]
+  );
+  const handleAllCommentChange = React.useCallback(
+    (commentIdVal: string) => {
+      const commentId = commentIdVal !== '' ? Number(commentIdVal) : undefined;
+      if (!commentId) {
+        setSlotComments([]);
+        return;
+      }
+      const allSlotComments: AddressCommentInput[] = [];
+      labware.slots.forEach((slot) => {
+        if (isSlotFilled(slot)) {
+          allSlotComments.push({ address: slot.address, commentId: commentId });
+        }
+      });
+      setSlotComments(allSlotComments);
+    },
+    [setSlotComments, labware]
+  );
+
+  const slotBuilder = (slot: SlotFieldsFragment): React.ReactNode => {
+    return (
+      isSlotFilled(slot) && (
+        <div className={`flex flex-col border-b border-gray-300`}>
+          <div className="w-full">
+            <Select
+              value={getComment(slot.address)}
+              emptyOption={true}
+              data-testid={'comment'}
+              onChange={(e) => handleCommentChange(slot.address, e.currentTarget.value)}
+            >
+              {optionValues(comments, 'text', 'id')}
+            </Select>
+          </div>
+        </div>
+      )
+    );
+  };
+  return (
+    <div>
+      {labware && (
+        <>
+          <div className="flex flex-row items-center justify-end">
+            {<RemoveButton data-testid={'remove'} type="button" onClick={() => removeLabware(labware.barcode)} />}
+          </div>
+          <div className="flex flex-col items-center justify-around">
+            {/* Display the layout of the labware */}
+            <div className="bg-blue-100" data-testid={'labware'}>
+              <Labware labware={labware} slotBuilder={slotBuilder} />
+            </div>
+
+            {/* Display the list of pass/fail comments */}
```

Comment:
What does pass/fail refer to here?

Reply:
That's a comment which is not updated, it is changed now to 'Display the dropdown to set/reset all comments'